### PR TITLE
Add copy button, logic, endpoint for saving multiple slides

### PR DESCRIPTION
**Pull request by ktbee**
@evmiguel @rwaldron here is my PR for adding forking functionality, let me know what you think of it! My proposed changes:

- Add "copy" button to the UI and pass some of the original scenario's data to the copy via `location.state`
- A `copyScenario` function that creates a new scenario with this passed data, gets the original slides and saves that slides data to the new scenario as well
- Add an endpoint for bulk saving slides
- Small amount of refactoring

### CODE REVIEW

File: client/components/Editor/index.jsx [hunk 1]
```diff
@@ -20,36 +20,35 @@ class Editor extends Component {
     constructor(props) {
         super(props);
 
-        if (!this.props.match.params.id) {
-            this.props.match.params.id = 'new';
```
Comment (ktbee) [inline client/components/Editor/index.jsx]: Changing the URL params data without changing the actual URL felt like a potential source of confusion in the future.

File: client/components/Editor/index.jsx [hunk 2]
```diff
@@ -144,29 +185,56 @@ class Editor extends Component {
...
+                };
+        }
+    }
+
     getSubmitCallback() {
         let endpoint, method;
         const scenarioId = this.props.match.params.id;
 
-        if (scenarioId === 'new') {
+        if (this.props.isNewScenario || scenarioId === 'copy') {
```
Comment (ktbee) [inline client/components/Editor/index.jsx]: We had this `isNewScenario` prop already thanks to our component in routes, I'm just taking advantage of it here.

File: client/components/Editor/index.jsx [hunk 3]
```diff
@@ -214,35 +277,28 @@ class Editor extends Component {
                 onClick={this.onClickScenarioAction}
             />
         );
+        const editTabMenu = Object.keys(this.state.tabs).map(tabType => {
```
Comment (ktbee) [inline client/components/Editor/index.jsx]: Switching to using the `tabs` state object saves some copy pasting and also needing to check if it's a new scenario or not.

File: client/components/ScenarioEditor/index.jsx [hunk 4]
```diff
@@ -14,8 +14,6 @@ class ScenarioEditor extends Component {
             categories: []
         };
 
-        this.fetchScenario = this.fetchScenario.bind(this);
```
Comment (ktbee) [inline client/components/ScenarioEditor/index.jsx]: I ended up finding that this `fetchScenario` is redundant with the one in the top level editor and was causing me some state issues when copying.

File: client/components/Editor/index.jsx (hunk 1 above)
Comment (rwaldron) [reply client/components/Editor/index.jsx]: I think this was a vestigial change that was no longer needed after you made the `<NewScenario>` wrapper.

Comment (ktbee) [reply client/components/Editor/index.jsx]: I actually found this logic helpful for figuring out the tabs we need later in the constructor and for setting state, so I think some form of it is worth keeping for now.

**Pull request merged**